Assets/MixedRealityToolkit.SDK/Features/Utilities/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,390 @@
+
+        [Tooltip("When lerping, use unscaled time. This is useful for apps that have a pause mechanism or otherwise adjust the game timescale.")]
+        public bool UseUnscaledTime = true;
+
+        [Tooltip("Lerp speed when moving focus point closer.")]
+        public float LerpStabilizationPlanePowerCloser = 4.0f;
+
+        [Tooltip("Lerp speed when moving focus point farther away.")]
+        public float LerpStabilizationPlanePowerFarther = 7.0f;
+
+        [SerializeField, Tooltip("Used to temporarily override the location of the stabilization plane.")]
+        private Transform targetOverride;
+        public Transform TargetOverride
+        {
+            get
+            {
+                return targetOverride;
+            }
+            set
+            {
+                if (targetOverride != value)
+                {
+                    targetOverride = value;
+                    if (targetOverride)
+                    {
+                        targetOverridePreviousPosition = targetOverride.position;
+                    }
+                }
+            }
+        }
+
+        [SerializeField, Tooltip("Keeps track of position-based velocity for the target object.")]
+        private bool trackVelocity;
+        public bool TrackVelocity
+        {
+            get
+            {
+                return trackVelocity;
+            }
+            set
+            {
+                trackVelocity = value;
+                if (TargetOverride)
+                {
+                    targetOverridePreviousPosition = TargetOverride.position;
+                }
+            }
+        }
+
+        [Tooltip("Default distance to set plane if plane is gaze-locked or if no object is hit.")]
+        public float DefaultPlaneDistance = 2.0f;
+
+        [Tooltip("Visualize the plane at runtime.")]
+        public bool DrawGizmos;
+        
+        [SerializeField, Tooltip("Override plane to use. Usually used to set plane to a slate like a menu")]
+        private StabilizationPlaneOverride overridePlane;
+
+        public StabilizationPlaneOverride OverridePlane
+        {
+            get
+            {
+                return overridePlane;
+            }
+            set
+            {
+                overridePlane = value;
+            }
+        }
+
+        private IMixedRealityInputSystem inputSystem = null;
+
+        /// <summary>
+        /// The active instance of the input system.
+        /// </summary>
+        private IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        /// <summary>
+        /// Position of the plane in world space.
+        /// </summary>
+        private Vector3 planePosition;
```

Comment:
nit: Worth maybe adding get to this? Like what is the current focus point?

Reply:
I think I won't add this for now since ideally applications wouldn't be trying to see the stabilization plane position at runtime